src/test/java/games/strategy/engine/data/TestGameDataMementoFactory.java

```diff
@@ -25,22 +22,21 @@ private TestGameDataMementoFactory() {}
    * @param value The property value.
    *
    * @return A new game data memento.
+   *
+   * @throws Exception If the game data memento cannot be created.
    */
-  public static Memento newMementoWithProperty(final String name, final @Nullable Object value) {
+  public static Memento newMementoWithProperty(final String name, final @Nullable Object value) throws Exception {
     checkNotNull(name);
 
     final Map<String, Object> propertiesByName = newValidMementoPropertiesByName();
     propertiesByName.put(name, value);
     return newMementoWithProperties(propertiesByName);
   }
 
-  private static Map<String, Object> newValidMementoPropertiesByName() {
-    final GameData gameData = TestGameDataFactory.newValidGameData();
-    return Maps.newHashMap(ImmutableMap.<String, Object>builder()
-        .put(GameDataMemento.PropertyNames.NAME, gameData.getGameName())
-        .put(GameDataMemento.PropertyNames.VERSION, gameData.getGameVersion())
-        // TODO: handle other properties
-        .build());
+  private static Map<String, Object> newValidMementoPropertiesByName() throws Exception {
```

Comment:
I refactored this method to avoid yet another location that had to be updated every time a new property is added to the `GameData` memento (it now piggybacks on the list that is maintained in `GameDataMemento$ExportHandler#exportProperties()`).  It relies on accessing the memento state through the implementation type `PropertyBagMemento`.  Thus, I added the `newExporterInternal()` package-private helpers to `GameDataMemento` to avoid exposing this type in the public API.